lib/matplotlib/tests/test_datetime.py

```diff
@@ -318,11 +318,40 @@ def test_spy(self):
         fig, ax = plt.subplots()
         ax.spy(...)
 
-    @pytest.mark.xfail(reason="Test for stackplot not written yet")
     @mpl.style.context("default")
     def test_stackplot(self):
-        fig, ax = plt.subplots()
-        ax.stackplot(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        N = 10
+        fig, (ax1, ax2) = plt.subplots(2, 1, constrained_layout=True)
+
+        stacked_nums = np.array([range(1, N)] * 4)
+
+        dates_days = np.array(
+            [datetime.datetime(2023, 1, i) for i in range(1, N)]
+        )
+        dates_years = np.array(
+            [datetime.datetime(1970 + i, 1, 1) for i in range(1, N)]
+        )
+
+        timedelta_days = np.array(
+            [i * datetime.timedelta(days=1) for i in range(1, N)]
+        )
+        timedelta_hours = np.array(
+            [i * datetime.timedelta(hours=24) for i in range(1, N)]
+        )
+        timedelta_seconds = np.array(
+            [i * datetime.timedelta(seconds=86400) for i in range(1, N)]
+        )
+
+        ax1.stackplot(dates_years, stacked_nums)
+
+        ax2.stackplot(
+            dates_days, dates_days, timedelta_days, timedelta_hours, timedelta_seconds
+        )
+        ax2.set_ylim(
+            datetime.datetime(2022, 12, 31),
+            datetime.datetime(2023, 2, 8)
+        )
```

Comment:
These limits are hiding the fact that the blue section goes all the way to 1970 because of the implicit 0 of `stackplot`.

I would support extending the API of the `baseline` kwarg to accept a scalar value, but that is not how it is implemented now, and I think that is out of the scope of this PR.

As it stands, I think I would rather _not_ test the "timedelta on the y axis" case than provide this test which is not a sensible use of `datetime` (though I think it is not _far_ off, and you made something that looked _okay_... I just think that the tools are not there to make using datetimes sensible here)

I do think `timedelta` is the right unit for this plot, to be clear... just it needs that extra explicit baseline, which is not currently supported (only zero or the various "wiggle" type options are supported currently)